File: source/src/beginner_tutorials/src/my_msg_talker.py
#!/usr/bin/env python
# license removed for brevity
import rospy
from beginner_tutorials.msg import my_msg


def talker():
    pub = rospy.Publisher('chatter', my_msg, queue_size=10)
    rospy.init_node('talker', anonymous=True)
    rate = rospy.Rate(10)  # 10hz
    count = 1
    while not rospy.is_shutdown():
        msg = my_msg()
        msg.id = count
        msg.title = "hello"
        msg.content = "hello from python"
        rospy.loginfo(msg.id)
        pub.publish(msg)
        count = count + 1
        rate.sleep()


if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: source/src/beginner_tutorials/src/my_msg_talker.cpp
#include "beginner_tutorials/my_msg.h"
#include "ros/ros.h"
int main(int argc, char **argv)
{
    ros::init(argc, argv, "talker");
    ros::NodeHandle n;
    ros::Publisher chatter_pub = n.advertise<beginner_tutorials::my_msg>("chatter", 1000);
    ros::Rate loop_rate(10);
    int count = 0;
    while (ros::ok())
    {
        beginner_tutorials::my_msg msg;
        msg.id = count;
        msg.title = "hello";
        msg.content = "hello from c++";
        ROS_INFO("%d", count); // stdout
        chatter_pub.publish(msg);
        ros::spinOnce();
        loop_rate.sleep();
        ++count;
    }
    return 0;
}